diff --git a/bootstrap.sh b/bootstrap.sh
index a206e0a..2e6439c 100755
--- a/bootstrap.sh
+++ b/bootstrap.sh
@@ -45,8 +45,8 @@ then
 fi
 
 package=wirecell
-origpkgver=v0_5_2
-pkgver=${origpkgver}c
+origpkgver=v0_6_0dev
+pkgver=${origpkgver}
 ssibuildshims_version=v0_16_00
 pkgdotver=`echo ${origpkgver} | sed -e 's/_/./g' | sed -e 's/^v//'`
 sourceurl=https://github.com/WireCell/wire-cell-build.git
@@ -85,8 +85,8 @@ fi
 cd ${pkgdir}/tar || ssi_die "could not cd ${pkgdir}/tar"
 git clone https://github.com/WireCell/wire-cell-build.git
 cd ${pkgdir}/tar/wire-cell-build || ssi_die "could not cd ${pkgdir}/tar/wire-cell-build"
-git co -b ${pkgver} ${pkgdotver}
-./switch-git-urls
+# git checkout -b ${pkgver} ${pkgdotver}
+# ./switch-git-urls
 git submodule init
 git submodule update
 
diff --git a/build_wirecell.sh b/build_wirecell.sh
index 1bce899..d4b7989 100755
--- a/build_wirecell.sh
+++ b/build_wirecell.sh
@@ -54,8 +54,8 @@ fi
 # -------------------------------------------------------------------
 
 package=wirecell
-origpkgver=v0_5_2
-pkgver=${origpkgver}c
+origpkgver=v0_6_0dev
+pkgver=${origpkgver}
 ssibuildshims_version=v0_16_00
 pkgdotver=`echo ${origpkgver} | sed -e 's/_/./g' | sed -e 's/^v//'`
 pkgtarfile=${package}-${pkgdotver}.tar.bz2
@@ -113,31 +113,31 @@ set -x
 cd ${pkgdir} || exit 1
 tar xf ${tardir}/${pkgtarfile} || exit 1
 # patch
-patch -b -p0 < ${patchdir}/wirecell.patch  || ssi_die "Failed to apply patch"
+# patch -b -p0 < ${patchdir}/wirecell.patch  || ssi_die "Failed to apply patch"
 
 cd ${pkgdir}/wire-cell-build || exit 1
 
 echo $PKG_CONFIG_PATH
 
 env CC=gcc CXX=g++ FC=gfortran ./wcb configure \
-      --fftw-no-single=true \
-      --with-jsoncpp $JSONCPP_FQ_DIR \
-      --with-tbb $TBB_FQ_DIR \
-      --with-eigen  $EIGEN_DIR \
-      --with-root $ROOTSYS \
-      --with-fftw $FFTW_FQ_DIR \
-      --with-fftw-include $FFTW_INC \
-      --with-fftw-lib $FFTW_LIBRARY \
-      --boost-includes $BOOST_FQ_DIR/include \
-      --boost-libs $BOOST_FQ_DIR/lib \
+      --with-jsoncpp=$JSONCPP_FQ_DIR \
+      --with-jsonnet=$JSONNET_FQ_DIR \
+      --with-tbb=$TBB_FQ_DIR \
+      --with-eigen=$EIGEN_DIR \
+      --with-root=$ROOTSYS \
+      --with-fftw=$FFTW_FQ_DIR \
+      --with-fftw-include=$FFTW_INC \
+      --with-fftw-lib=$FFTW_LIBRARY \
+      --boost-includes=$BOOST_FQ_DIR/include \
+      --boost-libs=$BOOST_FQ_DIR/lib \
       --boost-mt \
       --prefix="${pkgdir}"
 (( $? == 0 )) || ssi_die "wcb configure failed."
 
-./wcb build install || exit 1
+./wcb --notests build install || exit 1
 
-# run tests
-./wcb --alltests --testcmd="env LD_LIBRARY_PATH=$WCT_EXTERNALS/lib:`pwd`/install/lib %s" || ssi_die "tests failed"
+# run tests.  Note, wcb does not return failure if some tests fail.
+./wcb --alltests
 
 set +x
 
@@ -149,7 +149,8 @@ then
 fi
 
 # real ups declare
-${SSIBUILDSHIMS_DIR}/bin/declare_product ${product_dir} ${package} ${pkgver} ${fullqual}
+${SSIBUILDSHIMS_DIR}/bin/declare_product ${product_dir} ${package} ${pkgver} ${fullqual} || \
+  ssi_die "failed to declare ${package} ${pkgver} ${fullqual}"
 
 # -------------------------------------------------------------------
 # common bottom stuff
diff --git a/ups/wirecell.table b/ups/wirecell.table
index e915539..b1a33e8 100644
--- a/ups/wirecell.table
+++ b/ups/wirecell.table
@@ -13,6 +13,7 @@ Qualifiers = "e14:prof"
     setupRequired(boost v1_63_0b -q +e14:+prof)
     setupRequired(tbb v2017_3c -q +e14:+prof)
     setupRequired(jsoncpp v1_7_7 -q +e14:+prof)
+    setupRequired(jsonnet v0_9_3 -q +e14:+prof)
     setupRequired( gcc v6_3_0 )
 
 Flavor = ANY
@@ -26,6 +27,7 @@ Qualifiers = "e14:debug"
     setupRequired(boost v1_63_0b -q +e14:+debug)
     setupRequired(tbb v2017_3c -q +e14:+debug)
     setupRequired(jsoncpp v1_7_7 -q +e14:+debug)
+    setupRequired(jsonnet v0_9_3 -q +e14:+debug)
     setupRequired( gcc v6_3_0 )
 
 Flavor = ANY
@@ -39,6 +41,7 @@ Qualifiers = "e10:prof"
     setupRequired(boost v1_63_0b -q +e10:+prof)
     setupRequired(tbb v2017_3c -q +e10:+prof)
     setupRequired(jsoncpp v1_7_7 -q +e10:+prof)
+    setupRequired(jsonnet v0_9_3 -q +e10:+prof)
     setupRequired( gcc v4_9_3a )
 
 Flavor = ANY
@@ -52,6 +55,7 @@ Qualifiers = "e10:debug"
     setupRequired(boost v1_63_0b -q +e10:+debug)
     setupRequired(tbb v2017_3c -q +e10:+debug)
     setupRequired(jsoncpp v1_7_7 -q +e10:+debug)
+    setupRequired(jsonnet v0_9_3 -q +e10:+debug)
     setupRequired( gcc v4_9_3a )
 
 Common:
